src/providers/cryptoauthlib/hash.rs

```diff
@@ -33,4 +34,40 @@ impl Provider {
             }
         }
     }
+
+    pub(super) fn psa_hash_compare_internal(
+        &self,
+        op: psa_hash_compare::Operation,
+    ) -> Result<psa_hash_compare::Result> {
+        let op_hash = op.hash.to_vec();
```

Comment:
Do you need to use `to_vec`? I think since `Deref` is implement on those types you might not have to convert them to plain `Vec`. The good thing with keeping them in their original form is that they will get zeroized at the end.